src/engine/contactsengine.cpp

```diff
@@ -737,8 +769,21 @@ class JobContactReader : public ContactReader
 
 void JobThread::run()
 {
-    ContactsDatabase database;
-    if (!database.open(QString(QLatin1String("qtcontacts-sqlite-job-%1")).arg(m_databaseUuid), m_nonprivileged, true)) {
+    QString dbId(QStringLiteral("qtcontacts-sqlite%1-job-%2"));
+    dbId = dbId.arg(m_autoTest ? QStringLiteral("-test") : QString()).arg(m_databaseUuid);
+
+    QMutexLocker locker(&m_mutex);
+
+    m_database.open(dbId, m_nonprivileged, m_autoTest);
+    m_nonprivileged = m_database.nonprivileged();
+    m_running = true;
+
+    {
+        MutexUnlocker unlocker(locker);
+        m_wait.wakeOne();
+    }
```

Comment:
Ok. I thought I saw a race but now that I try to describe it I think it can't happen.